src/drivers/mailbox/bcm2835_mailbox_property.c

```diff
@@ -0,0 +1,284 @@
+        case TAG_GET_PITCH:
+        case TAG_BLANK_SCREEN:
+            pt[pt_index++] = 4;
+            pt[pt_index++] = 0; /* Request */
+
+            if( ( tag == TAG_SET_DEPTH ) ||
+                ( tag == TAG_SET_PIXEL_ORDER ) ||
+                ( tag == TAG_SET_ALPHA_MODE ) ||
+                ( tag == TAG_BLANK_SCREEN ) )
+            {
+                /* Colour Depth, bits-per-pixel \ Pixel Order State / on/off */
+                pt[pt_index++] = va_arg( vl, int );
+            }
+            else
+            {
+                pt_index += 1;
+            }
+            break;
+
+        case TAG_GET_OVERSCAN:
+        case TAG_SET_OVERSCAN:
+            pt[pt_index++] = 16;
+            pt[pt_index++] = 0; /* Request */
+
+            if( ( tag == TAG_SET_OVERSCAN ) )
+            {
+                pt[pt_index++] = va_arg( vl, int ); /* Top pixels */
+                pt[pt_index++] = va_arg( vl, int ); /* Bottom pixels */
+                pt[pt_index++] = va_arg( vl, int ); /* Left pixels */
+                pt[pt_index++] = va_arg( vl, int ); /* Right pixels */
+            }
+            else
+            {
+                pt_index += 4;
+            }
+            break;
+
+        case TAG_RELEASE_BUFFER:
+            pt[pt_index++] = 0; /* no arguments */
+            pt[pt_index++] = 0; /* Request */
+            break;
+
+        default:
+            /* Unsupported tags, just remove the tag from the list */
+            pt_index--;
+            break;
+    }
+
+    /* Make sure the tags are 0 terminated to end the list and update the buffer size */
+    pt[pt_index] = 0;
+
+    va_end( vl );
+}
+
+int bcm2835_property_process( void )
+{
+    int result;
+
+    /* Fill in the size of the buffer */
+    pt[PT_OSIZE] = ( pt_index + 1 ) << 2;
+    pt[PT_OREQUEST_OR_RESPONSE] = 0;
+
+    bcm2835_mailbox_write((unsigned int)pt, BCM2835_TAGS_ARM_TO_VC);
+    
+    result = bcm2835_mailbox_read( BCM2835_TAGS_ARM_TO_VC );
+
+    return result;
+}
+
+bcm2835_mailbox_property_t* bcm2835_property_get( bcm2835_mailbox_tag_t tag )
+{
+    static bcm2835_mailbox_property_t property;
+    int* tag_buffer = 0;
+
+    property.tag = tag;
+
+    /* Get the tag from the buffer. Start at the first tag position  */
+    int index = 2;
+
+    while( index < ( pt[PT_OSIZE] >> 2 ) )
+    {
+        /* printf( "Test Tag: [%d] %8.8X\r\n", index, pt[index] ); */
+        if( pt[index] == tag )
+        {
+            tag_buffer = &pt[index];
+            break;
+        }
+
+        /* Progress to the next tag if we haven't yet discovered the tag */
+        index += ( pt[index + 1] >> 2 ) + 3;
+    }
+
+    /* Return NULL of the property tag cannot be found in the buffer */
+    if( tag_buffer == 0 )
+        return 0;
+
+    /* Return the required data */
+    property.byte_length = tag_buffer[T_ORESPONSE] & 0xFFFF;
+    memcpy( property.data.buffer_8, &tag_buffer[T_OVALUE], property.byte_length );
+
+    return &property;
+}
```

Comment:
Could you please add a new line at the and of the file

Reply:
newlines are there, as per what I see in my editor.  But oddly, sometimes, they don't show in a git DIFF view.